src/main/java/com/avispl/symphony/dal/communicator/lg/lcd/LgLCDDevice.java

```diff
@@ -114,55 +204,439 @@ public void controlProperties(List<ControllableProperty> controllableProperties)
+						stats.put(group + LgLCDConstants.SCREEN_COLOR_VALUE, String.valueOf(screenColor));
+						break;
+					case SHARPNESS:
+						int sharpness = (int) Float.parseFloat(value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), Integer.toHexString(sharpness).getBytes(StandardCharsets.UTF_8));
+						stats.put(group + LgLCDConstants.SHARPNESS_VALUE, String.valueOf(sharpness));
+						break;
+					case TINT:
+						int tint = (int) Float.parseFloat(value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), Integer.toHexString(tint).getBytes(StandardCharsets.UTF_8));
+						stats.put(group + LgLCDConstants.TINT_VALUE, String.valueOf(tint));
+						break;
+					case ASPECT_RATIO:
+						String aspectRatio = EnumTypeHandler.getValueOfEnumByName(AspectRatio.class, value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), aspectRatio.getBytes(StandardCharsets.UTF_8));
+						break;
+					case BRIGHTNESS_SIZE:
+						String brightnessSize = EnumTypeHandler.getValueOfEnumByName(BrightnessSize.class, value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), brightnessSize.getBytes(StandardCharsets.UTF_8));
+						break;
+					case LANGUAGE:
+						String language = EnumTypeHandler.getValueOfEnumByName(Language.class, value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), language.getBytes(StandardCharsets.UTF_8));
+						break;
+					case SOUND_MODE:
+						String soundMode = EnumTypeHandler.getValueOfEnumByName(SoundMode.class, value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), soundMode.getBytes(StandardCharsets.UTF_8));
+						break;
+					case PICTURE_MODE:
+						String pictureMode = EnumTypeHandler.getValueOfEnumByName(PictureMode.class, value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), pictureMode.getBytes(StandardCharsets.UTF_8));
+						break;
+					case POWER_ON_STATUS:
+						String powerStatus = EnumTypeHandler.getValueOfEnumByName(PowerStatus.class, value);
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), powerStatus.getBytes(StandardCharsets.UTF_8));
+						break;
+					case NO_IR_POWER_OFF:
+					case NO_SIGNAL_POWER_OFF:
+						String powerValue = String.valueOf(LgLCDConstants.ZERO) + LgLCDConstants.ZERO;
+						if (String.valueOf(LgLCDConstants.NUMBER_ONE).equals(value)) {
+							powerValue = String.valueOf(LgLCDConstants.ZERO) + LgLCDConstants.NUMBER_ONE;
+						}
+						sendRequestToControlValue(lgControllingCommand.getCommandNames(), powerValue.getBytes(StandardCharsets.UTF_8));
+						break;
+					default:
+						logger.debug(String.format("Property name %s doesn't support", propertyKey));
+				}
 			}
-			throw e;
+			updateValueForTheControllableProperty(property, value, stats, advancedControllableProperties);
+			destroyChannel();
```

Comment:
do you really need to destroy the channel every time after the control operation?